File: FormacaoPhp/PHPAvancOOP2/php/classes/funcionarios/Diretor.php
<?php

namespace classes\funcionarios;

use classes\abstratas\FuncionarioAutenticavel;

class Diretor extends FuncionarioAutenticavel
{
    public $gerenciados;
    public function getBonificacao()
    {
        return $this->salario * 0.5;
    }

    public function gerenciados(){
        return json_encode($this->gerenciados);
    }
    
    public function __toString()
    {
        return json_encode([$this->nome, $this->cpf, $this->salario]);
    }
}

File: FormacaoCSharp/FormacaoCSharp/Parte3/P3_ByteBank/Funcionarios/Diretor.cs
﻿using P3_ByteBank.Sistemas;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace P3_ByteBank.Funcionarios
{
    public class Diretor : FuncionarioAutenticavel
    {
        public Diretor(string cpf) : base(cpf: cpf, salario: 5000)
        {
        }

        public override void AumentarSalario()
        {
            Salario *= 1.15;
        }

        public override double GetBonificacao()
        {
            return Salario * 0.5;
        }

        ~Diretor() { Console.WriteLine("Destruido " + Nome); }
    }
}
